packages/shared/src/transactions.ts

```diff
@@ -139,3 +144,10 @@ export const extractTokenIds = (tokenIds: Asset['id'][], ios: Transaction['input
     })
   })
 }
+
+export const findTransactionReferenceAddress = (addresses: AddressHash[], tx: Transaction | PendingTransaction) =>
+  addresses.find((address) => isAddressPresentInInputsOutputs(address, tx))
+
```

Comment:
We should test this thoroughly. I'm not saying it doesn't work, but there were many edge cases in the past so I'm careful :) Like what happens for an internal transfer, or some weird contract operations.

Reply:
This is the same code that we used to have (see `getTransactionsOfAddress` above), but yes, I agree.